plugins/modules/foreman_template_sync.py

```diff
@@ -0,0 +1,220 @@
+      - On I(direction=export) export templates with names matching this regex.
+      - Case-insensitive, snippets are not filtered.
+    required: false
+    type: str
+  negate:
+    description: Negate the prefix (for purging).
+    required: false
+    type: bool
+  dirname:
+    description: The directory within Git repo containing the templates.
+    required: false
+    type: str
+  locations:
+    description: REPLACE locations with given list.
+    required: false
+    type: list
+    elements: str
+  organizations:
+    description: REPLACE organizations with given list.
+    required: false
+    type: list
+    elements: str
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: Sync templates from git repo
+  foreman_template_sync:
+    repo: https://github.com/theforeman/community-templates.git
+    branch: 1.24-stable
+    associate: new
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "changeme"
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+
+
+def main():
+    module = ForemanEntityAnsibleModule(
+        entity_spec=dict(
+            direction=dict(choices=['import', 'export'], default='import'),
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
+            associate=dict(choices=['always', 'new', 'never']),
+            metadata_export_mode=dict(choices=['refresh', 'remove', 'keep']),
+            prefix=dict(),
+            branch=dict(),
+            repo=dict(),
+            filter=dict(),
+            dirname=dict(),
+            verbose=dict(type='bool'),
+            force=dict(type='bool'),
+            lock=dict(type='bool'),
+            negate=dict(type='bool'),
+            locations=dict(type='entity_list', flat_name='location_ids'),
+            organizations=dict(type='entity_list', flat_name='organization_ids'),
+        ),
+        supports_check_mode=False,
+    )
+
+    entity_dict = module.clean_params()
+    module.connect()
+
+    if 'template' not in module.foremanapi.resources:
+        raise Exception('The server does not seem to have the foreman_templates plugin installed.')
+
+    all_templates = []
+
+    template_types = ['provisioning_templates', 'report_templates', 'ptables']
+    if 'job_templates' in module.foremanapi.resources:
+        template_types.append('job_templates')
+    for template_type in template_types:
+        resources = module.list_resource(template_type)
+        for resource in resources:
+            all_templates.append([resource['name'], resource['id']])
+
+    result = module.resource_action('template', entity_dict['direction'], record_change=False, params=entity_dict)
+    changed = False
+    msg_templates = result['message'].pop('templates', [])
+
+    if entity_dict['direction'] == 'import':
+        diff = {'changed': [], 'new': []}
+    else:
```

Comment:
I think, the flow of control here is a little bit confusing. Can you move this else-branch to the end of the method?

Reply:
totally agree, doing